server/src/main/java/org/opensearch/common/network/NetworkModule.java

```diff
@@ -263,15 +270,15 @@ public Supplier<Transport> getTransportSupplier() {
      * Registers a new {@link TransportInterceptor}
      */
     private void registerTransportInterceptor(TransportInterceptor interceptor) {
```

Comment:
Currently all the updations are not exposed as public methods in networkModule. 
We can make this method public and add admissionControlInterceptor to the list but it will be added at the end of the plugin interceptors. So our admissionControlInterceptor will execute after all plugin interceptors.

@bharath-techie @psychbot @anshu1106 @Bukhtawar Need thoughts on below queries to conclude on final approach:

1. Can we add admissionControl interceptor after plugin interceptor -> IMO as there is no point of allowing request to flow if there is possibility of rejection.
2. If we want to add before the plugin interceptors the approaches are:
-    Create one registry to add all coreTransportInterceptor and pass to that into networkModule and register in constructor (which is present in current PR).
- Expose `registerTransportInterceptor` method as static and make the `transportInterceptors` List as static variable and use it in `Node.java` to add new interceptors.
- One more hacky way is expose one more public method to registerInterceptors before the plugin interceptors. As this can done once the networkModule initialised so we need to update entire list again.

Reply:
For me, option 1 - present code still looks clean compared to other options.
For option 3 , we can add core interceptor to start of the list :
```
public void registerCoreTransportInterceptor(TransportInterceptor interceptor) {
             this.transportIntercetpors.add(0, Objects.requireNonNull(interceptor, "interceptor must not be null"));
}
```

But again the order of core interceptors will be based on the caller order.

Reply:
> But again the order of core interceptors will be based on the caller order.

Yes but in any approach the order of interceptors will be based on caller order only

Reply:
Updated the PR with the option 3. we will expose one public method to add core Interceptors first.

Reply:
rather than having an option to either run core interceptors first or plugin interceptors, is it possible to define an execution order in the interceptor class? 

Another thought, won't running interceptors serially add to request latency further?

Reply:
> rather than having an option to either run core interceptors first or plugin interceptors, is it possible to define an execution order in the interceptor class?

when the network module is loaded all the plugins which are implemented networkPlugin will be added into interceptors list in the constructor. So far we have only transportInterceptor from security plugin.  As all plugins interceptors are loaded once we can add our interceptor either before or after plugin interceptors. 

>  Another thought, won't running interceptors serially add to request latency further?
 
The execution of the interceptors are done serially by design. As there are only two interceptors now there won't be much latency added. The interceptors are very light weight and we will ensure the logic in the handler will be executed in O(1) time complexity.

Reply:
How about passing a List of interceptors in the ctor of NetworkModule in this case AdmissionControlInterceptor and ensure it gets added first?